react_app/src/reducers/game.reducers.js

```diff
@@ -0,0 +1,40 @@
+import * as ActionTypes from "actions/actionTypes";
+
+const initialState = {
+  teamName: "",
+  teamScore: 0,
+  teamRank: false,
+  viewingChallenge: null
+};
+
+/**
+ * action creators
+ */
+
+/**
+ * reducers
+ */
+/* state param takes default of initialState if not otherwise set */
+export default (state = initialState, action) => {
+  switch (action.type) {
+    case ActionTypes.GET_GAME_STATE:
+      return {
+        ...state,                               // unpack state
+        teamName: action.response.teamName,     // override those things that are set
+        teamScore: action.response.teamScore,
+        teamRank: action.response.teamRank,
+        viewingChallenge: null
+      };
+    case ActionTypes.OPEN_CHALLENGE:
+        console.log("opening challenge... " + action.challengeID)
+        return {
+            viewingChallenge: action.challengeID
```

Comment:
This is going to blat the rest of your state because you're only returning one key - is that intentional? 

Also this `viewingChallenge` is another candidate to be put into a router.

Reply:
No, that wasn't intentional, but shouldn't it only kill the state.game portions? Or will it nuke the auth section too?

Reply:
Fixed by unpacking state in each resolver before overwriting specific keys